hail/src/main/scala/is/hail/expr/ir/Emit.scala

```diff
@@ -885,6 +892,77 @@ class Emit[C](
         val AggContainer(_, sc, _) = container.get
         presentC(sc.states(i).serializeToRegion(cb, coerce[PBinary](pt), region))
 
+      case x@ShuffleWith(
+        keyFields,
+        rowType,
+        rowEType,
+        keyEType,
+        name,
+        writerIR,
+        readersIR
+      ) =>
+        val shuffleType = x.shuffleType
+        val shufflePType = x.shufflePType
+
+        val shuffle = CodeShuffleClient.createValue(cb, mb.ecb.getType(shuffleType))
+
+        cb.append(shuffle.start())
+
+        val uuid = PCanonicalShuffleSettable.fromArrayBytes(
+          cb, region, shufflePType, shuffle.uuid())
+
+        val shuffleEnv = env.bind(name -> mb.newPresentEmitSettable(uuid.pt, uuid))
+
+        val successfulShuffleIds: PValue = emitI(writerIR, env = shuffleEnv).consume(cb,
```

Comment:
Yes, see comment below. Eventually this will be used to ensure that the shuffler only preserves the one attempt per task. That functionality is not yet implemented.